### PR TITLE
`azurerm_application_gateway` - support for the `globalConfiguration` property

### PR DESCRIPTION
**Pull request by ms-zhenhua**
fix https://github.com/hashicorp/terraform-provider-azurerm/issues/17627
image

### CODE REVIEW

File: internal/services/network/application_gateway_resource.go [hunk 1]
```diff
@@ -409,6 +409,24 @@ func resourceApplicationGateway() *pluginsdk.Resource {
 				},
 			},
 
+			"global_configuration": {
```
Comment (katbyte) [inline internal/services/network/application_gateway_resource.go:412]: configuration is redundant
```suggestion
			"global": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
1 minor comment but otherwise looks good

**Comment by ms-zhenhua** [comment]
Hi @katbyte, thank you for reviewing my code. I have resolved the comments and rerun the testcase. Kindly help review it again.

File: internal/services/network/application_gateway_resource.go (hunk 1 above)
Comment (markwaddle) [reply internal/services/network/application_gateway_resource.go]: i recommend reverting this change, although it is not a blocker for me.

the name `global_configuration` matches the nomenclature used by Azure resource manager (`globalConfiguration`). adopting their nomenclature, even if somewhat redundant, seems useful for users of this resource. it also matches the convention established by other `*_configuration`s, such as `frontend_ip_configuration` (`frontendIPConfiguration` in ARM)

using the name `global_configuration` keeps the naming consistent with ARM and consistent within this resource

**Comment by markwaddle** [comment]
@katbyte  @ms-zhenhua i recommend reverting the name change of `global_configuration` -> `global`. i appreciate the desire to remove redundancy in names. i recommend we consider the fact that this name is established by ARM. and that there is already a convention in this resource to match the ARM names. i personally find it useful when hopping between the two. https://docs.microsoft.com/en-us/azure/templates/microsoft.network/applicationgateways?tabs=bicep&pivots=deployment-language-bicep#resource-format

the name `global_configuration` matches the nomenclature used by Azure resource manager (`globalConfiguration`). it also matches the convention established by other `*_configuration`s, such as `frontend_ip_configuration` (`frontendIPConfiguration` in ARM)

using the name `global_configuration` keeps the naming consistent with ARM and consistent within this resource

**Review by katbyte — APPROVED** [review]
Thanks @ms-zhenhua - LGTM 

@markwaddle - searching for api terms will already not match the property names as the api is camel case vs terraforms snake case, there is a disconnect between terraform and APIs already. one that i'd like to solve in the future by amending documentation with api names somehow once we are generating docs. either way Going forward we will be removing configuration,info, config, etc from block names.

**Pull request merged**